tasks/terraform-cli/task.json

```diff
@@ -318,6 +335,30 @@
             "helpMarkDown": "Skip attempting to create existing workspaces",
             "visibleRule": "workspaceSubCommand = new"
         },
+        {
+            "name": "stateMoveSource",
+            "type": "string",
+            "label": "State Move Source",
+            "required": false,
+            "visibleRule": "stateSubCommand = mv",
+            "helpMarkDown": "The name of the source resource"
+        },
+        {
+            "name": "stateMoveDestination",
+            "type": "string",
+            "label": "State Move Destination",
+            "required": false,
+            "visibleRule": "stateSubCommand = mv",
+            "helpMarkDown": "The name of the destination resource"
+        },
+        {
+            "name": "stateAddresses",

```

Comment:
@charleszipp Not sure about this spec as I'm using `tasks.getDelimitedInput` to get this value

Reply:
I havent tried `getDelimitedInput` before so not sure either. Checking the [schema for task.json](https://github.com/microsoft/azure-pipelines-task-lib/blob/b901effb8eaf6455b3044d57b0e0b4121aa7aa30/tasks.schema.json#L179), the only two that make sense are `string` and `multiline`.

We should probably add a sample pipeline to `pipelines/test` to test these for real from a pipeline before merge. This will ensure the PR CI executes the new commands as the users would. The `pipelines/tests/switch_workspaces.yml` would be a good example to follow. You just have to create the yml file for the test, then add it to the `pipelines-beta.yml` to ensure it gets executed as part of PR CI.

Additionally, you can run the task locally. There are instructions for doing this in the `tasks/terraform-cli/README.md` doc. You basically need to create a .env then run npm install && npm start. You will need to run it once to execute terraform init in the `./templates` directory. Then you can run again to execute the state commands

Sample init .env

```text
AGENT_TOOLSDIRECTORY=./../../_test-agent/tools
AGENT_TEMPDIRECTORY=./../../_test-agent/temp
SYSTEM_TEAMFOUNDATIONCOLLECTIONURI=https://dev.azure.com/my-org
SYSTEM_TEAMPROJECT=my-project

INPUT_AIINSTRUMENTATIONKEY=foo
INPUT_ALLOWTELEMETRYCOLLECTION=false
INPUT_COMMAND=init
INPUT_WORKINGDIRECTORY=./../../templates
INPUT_BACKENDTYPE=local
```

The following is a .env I used for testing the workspace commands. Its probably a good starting point for setting one up to test state commands

```text
AGENT_TOOLSDIRECTORY=./../../_test-agent/tools
AGENT_TEMPDIRECTORY=./../../_test-agent/temp
SYSTEM_TEAMFOUNDATIONCOLLECTIONURI=https://dev.azure.com/my-org
SYSTEM_TEAMPROJECT=my-project

INPUT_AIINSTRUMENTATIONKEY=foo
INPUT_ALLOWTELEMETRYCOLLECTION=false
INPUT_COMMAND=workspace
INPUT_WORKINGDIRECTORY=./../../templates

INPUT_WORKSPACESUBCOMMAND=new
INPUT_WORKSPACENAME=bar
```

Reply:
Very good point, I've tested with the `.env` procedure, worked like a charm.

I've added a new resource to the templates folder, that way the state pipeline can make test no impacting an Azure Resources. I did that to ensure that no resource group got  undeleted with the destroy